documentation/user-consent.md

```diff
@@ -48,122 +43,269 @@ The document details aspects regarding the consent management, which includes fo
+    Authorization: Basic {Credentials}
+    Content-Type: application/x-www-form-urlencoded
 
->[TO BE EDITED/COMPLETED] - Consent API: API to manage user consent, if required. For example when consent is captured by a third party and stored in operator. A third-party can register the consent captured from a user for a specific purpose.
+    login_hint=tel%3A%2B346xxyyyzzz&
+    … 
+    ```
+- Purpose under which the personal data associated to API consumption will be processed.
 
-## Annex: Flow source code
+The operator OGW platform will:
+
+- Validate user identifier, map it to a telco identifier if applicable, e.g.: map IP to MSISDN. Set the OAuth sub to the unique user id in the operator (Step 2).
+
+- Check if a user consent is needed, which depends on the legal basis associated to the purpose (“legitimate interest”, “contract”, “consent”, etc). If needed, it will check in operator consent master if user consent was already given for that identifier and for the requested purpose(s) (Steps 3-4).
+
+Then, two alternatives may occur.
+
+**Scenario 1**: User consent is not needed or consent is already granted (Step 5). OGW platform will directly return a 200 OK response with the CIBA auth request identifier (auth_req_id=OperatorAuthReqId) to the API invoker. This is a unique identifier to identify the authentication request made by the invoker.
+
+**Scenario 2**: Consent is needed and is not granted by the user yet (Step 6)
+
+- A mechanism is triggered to capture user consent in the operator:
+  
+  - The operator triggers an out-of-band consent capture mechanism to interact with the user. **Operators can choose the consent capture mechanism that best suits their capabilities, preferences and needs**. This mechanism can be a push notification, an SMS, etc.
+  - In parallel, the OGW platform operator returns a 200 OK response with the CIBA auth request identifier (auth_req_id=OperatorAuthReqId) to the API invoker to indicate that the authentication request has been accepted and is going to be processed (Step 6).
+
+<br>
+
+Then, the API invoker polls the token endpoint by making an "HTTP POST" request by sending the grant_type (`urn:openid:params:grant-type:ciba`) and auth_req_id (OperatorAuthReqId) parameters (Step 7).
+- The API invoker will receive the following error code if the authorization is still pending to be accepted or rejected by the user.
+    
+    ```
+    HTTP/1.1 400 Bad Request
+    Content-Type: application/json
+    
+    {
+        "error": "authorization_pending",
+    }
+    ```
+- When this response is received, the API invoker must wait the seconds of the `interval` value received in the CIBA authorization endpoint and then repeat the request until a final response is received.
+- Once the user has granted consent, the OGW platform operator will provide the access token (OperatorAccessToken) to the API invoker (Step 8). The OperatorAccessToken issued will be encrypted so no relevant information is disclosed.
```

Comment:
```suggestion
- Once the user has granted consent, the Operator's API Gateway Platform will provide the access token (OperatorAccessToken) to the API Invoker (Step 8). The OperatorAccessToken issued will be encrypted so no relevant information is disclosed.
```

Reply:
Fixed in https://github.com/camaraproject/IdentityAndConsentManagement/pull/54/commits/c7f9715fa8c597e4c9f94e8f0bbf3aecc501ead5.